squad-server/plugins/auto-kick-unassigned.js

```diff
@@ -0,0 +1,225 @@
+import BasePlugin from './base-plugin.js';
+import Logger from 'core/logger';
+
+export default class AutoKickUnassigned extends BasePlugin {
+  static get description() {
+    return 'The <code>AutoKickUnassigned</code> plugin will automatically kick players that are not in a squad after a specified ammount of time.';
+  }
+
+  static get defaultEnabled() {
+    return true;
+  }
+
+  static get optionsSpecification() {
+    return {
+      adminList: {
+        required: true,
+        description: 'List of server admins.',
+        connector: 'remoteAdminLists',
+        default: 'remoteAdminLists'
+      },
+      warningMessage: {
+        required: false,
+        description: 'Message SquadJS will send to players warning them they will be kicked',
+        default: 'Join a squad, you are are unassigned and will be kicked'
+      },
+      kickMessage: {
+        required: false,
+        description: 'Message to send to players when they are kicked',
+        default: 'Unassigned - automatically removed'
+      },
+      frequencyOfWarnings: {
+        required: false,
+        description: 'How often in seconds should we warn the player about being unassigned?',
+        default: 30
+      },
+      unassignedTimer: {
+        required: false,
+        description: 'How long in minutes to wait before a player that is unassigned is kicked',
+        default: 6
+      },
+      playerThreshold: {
+        required: false,
+        description:
+          'Player count required for AutoKick to start kicking players to disable set to -1 to disable',
+        default: 93
+      },
+      queueThreshold: {
+        required: false,
+        description:
+          'The number of players in the queue before AutoKick starts kicking players set to -1 to disable',
+        default: -1
+      },
+      roundStartDelay: {
+        required: false,
+        description:
+          'Time delay in minutes from start of the round before AutoKick starts kicking again',
+        default: 15
+      },
+      ignoreAdmins: {
+        required: false,
+        description: 'Whether or not admins will be auto kicked for being unassigned',
+        default: false
+      },
+      ignoreWhitelist: {
+        required: false,
+        description: 'Whether or not players in the whitelist will be auto kicked for being unassigned',
+        default: false
+      }
+    };
+  }
+
+  /**
+   * trackedPlayers[<steam64ID>] = <tracker>
+   *
+   *  <tracker> = {
+   *         player: <playerObj>
+   *       warnings: <int>
+   *      startTime: <Epoch Date>
+   *    warnTimerID: <intervalID>
+   *    kickTimerID: <timeoutID>
+   *  }
+   */
+  constructor(server, options) {
+    super();
+
+    this.server = server;
+    this.options = options;
```

Comment:
If you extend `BasePlugin`, just do `super(server, options, rawOptions)` and scrap the last two lines.